Library/Homebrew/install.rb

```diff
@@ -29,19 +29,44 @@ def perform_build_from_source_checks(all_fatal: false)
       Diagnostic.checks(:build_from_source_checks, fatal: all_fatal)
     end
 
-    def check_prefix
-      if Hardware::CPU.intel? && HOMEBREW_PREFIX.to_s == HOMEBREW_MACOS_ARM_DEFAULT_PREFIX
-        odie "Cannot install in Homebrew on Intel processor in ARM default prefix (#{HOMEBREW_PREFIX})!"
-      elsif Hardware::CPU.arm? && HOMEBREW_PREFIX.to_s == HOMEBREW_DEFAULT_PREFIX
-        odie <<~EOS
-          Cannot install in Homebrew on ARM processor in Intel default prefix (#{HOMEBREW_PREFIX})!
-          Please create a new installation in #{HOMEBREW_MACOS_ARM_DEFAULT_PREFIX} using one of the
-          "Alternative Installs" from:
-            #{Formatter.url("https://docs.brew.sh/Installation")}
-          You can migrate your previously installed formula list with:
-            brew bundle dump
+    def check_flavour_matches_architecture
+      homebrew_flavour = HOMEBREW_REPOSITORY.cd do
+        Utils.popen_read("git", "config", "--get", "homebrew.flavour").chomp.presence
```

Comment:
Could consider passing this as a global variable, too, so it only needs read once.

Reply:
I did that in my first draft, which:

> [introduced] a read-only HOMEBREW_FLAVOUR environment variable that tells us under which architecture a given Homebrew installation can be run.

I removed it after your comment that said:

> I'd like to avoid user configuration here until we feel it's needed from user issues/requests/confusion.

That environment variable wasn’t user configuration. It was read-only, would be set by Homebrew, ignoring whatever its initial value is. Its purpose was to pass it as a global variable so it only needs to be read once. I removed it anyway according to your request.

So you’d prefer me to add that variable back, right?

Reply:
> That environment variable wasn’t user configuration. It was read-only, would be set by Homebrew, ignoring whatever its initial value is. Its purpose was to pass it as a global variable so it only needs to be read once. I removed it anyway according to your request.

I didn't request you remove it, I requested we avoid it being user configurable as it wasn't clear from the proposal whether it would be or not.

> So you’d prefer me to add that variable back, right?

Yes, I think it would be worth considering if it aligns with the fix for https://github.com/Homebrew/brew/pull/9126#discussion_r524219514

https://github.com/Homebrew/brew/pull/9126#discussion_r524217249 is another global variable that should be passed from Bash into Ruby to avoid duplication if it's needed there.

Reply:
@MikeMcQuaid Ok, thanks. Will look for the old commit, restore it and apply your suggestions.

Reply:
Thanks!